### ISSUE TITLE
Is it possible to get the database name from the ActiveAndroid object?

### ISSUE THREAD

**Issue by joshuapinter** [pardom-zz/ActiveAndroid#153]
I know I can get it from the AndroidManifest metadata but can this be obtained from ActiveAndroid directly, such as through the Configuration class?

Thanks. 

Josh

**Comment by SleeplessByte** [comment]
The `Configuration` class does have a public method `Configuration#getDatabaseName` and it will be filled by `getMetaDataDatabaseNameOrDefault` if you don't set it when you call `Configuration#create`.

**Comment by joshuapinter** [comment]
Hmmm... I see that but how would you get access to it? 

Since it's not a class method you need to get access to the configuration instance.

**Comment by SleeplessByte** [comment]
@joshuapinter The eassiest way would be to initialize active Android like this:

``` Java
Configuration configuration = new Configuration.Builder( application ).create();
ActiveAndroid.initialize( configuration, false );
```

You can then access the shizzle through the `configuration` variable.
If you want to override the Manifest (or not use it):

``` Java
Configuration configuration = new Configuration.Builder( application )
    .setDatabaseName( DATABASE_NAME )
    .setDatabaseVersion( DATABASE_VERSION )
    .create();
ActiveAndroid.initialize( configuration, false );
```

**Comment by joshuapinter** [comment]
@SleeplessByte Right, but once `ActiveAndroid` is initialized, there's no way to get access to it. 

So, what I did and I think the best method to get the database name programatically after `ActiveAndroid` is initialized is to get it from the meta data, like so:

``` java
// Get Database name from manifest file.
String activeAndroidDatabaseName = getMetaData( activity.getApplication(), "AA_DB_NAME" ).toString();

public Object getMetaData(Application application, String name) {
    final ApplicationInfo applicationInfo;

    try {
      PackageManager packageManager = application.getApplicationContext().getPackageManager();
      String         packageName    = application.getApplicationContext().getPackageName();
      applicationInfo = packageManager.getApplicationInfo( packageName, PackageManager.GET_META_DATA );
    } 
    catch (PackageManager.NameNotFoundException e) { e.printStackTrace(); }

    return ( applicationInfo.metaData != null ? applicationInfo.metaData.get(name) : null );
}
```

**Comment by SleeplessByte** [comment]
@joshuapinter actually, that's not the best way, as you can programatically SET the database name. 

Apart from that, `ActiveAndroid` does provide a util method: 

``` Java
ReflectionUtils.getMetaData(Context context, String name)
```

You could actually do this with Reflection if you wanted to actually get the value. Just add the appropiate try catches, I am doing this by heart:

``` Java
Field dhField = com.activeandroid.Cache.class.getDeclaredField( "sDatabaseHelper" );
dhField.setAccessible( true );
SQLiteOpenHelper helper = (SQLiteOpenHelper) dhField.get( null );

String databaseName = helper.getDatabaseName();
```

**Edit** This will ONLY work **after** initialization.

**Comment by joshuapinter** [comment]
Oh nice. Okay, thanks for that. I appreciate the follow-up.

**Comment by joshuapinter** [comment]
@SleeplessByte No idea why I was so stubborn five years ago, as you were trying to lead me down the correct path. 

I've finally had a need to revisit this and here is my much better solution to getting the database name directly from the `ActiveAndroid` instance: 

```java
import org.apache.commons.io.FilenameUtils;

String databaseName = FilenameUtils.getName( ActiveAndroid.getDatabase().getPath() );
```

Much cleaner. 

I needed to do this because we stopped setting the database name in `AndroidManifest.xml` via `AA_DB_NAME` and instead dynamically configure and instantiate it in the app using `.setDatabaseName()`.

Anyway, five years on and just wanted to say you were right all along and thank you for your help and patience. 🙏
